core/src/services/azblob/core.rs

```diff
@@ -243,12 +245,19 @@ impl AzblobCore {
 
         let mut req = Request::put(&url);
 
+        if let Some(user_metadata) = args.user_metadata() {
```

Comment:
I'm setting the metadata headers only for the `azblob_put_blob_request`.

Should this also be done for `block` requests, for example, in `azblob_put_block_request`? Or the metadata will be only supported for `blobs`?

Reply:
I see that in `OSS` the metadata headers is also set for the `append_object_request`. Should the azure append blob requests also send these headers? `azblob_append_blob_request`, `azblob_init_appendable_blob_request`, etc..?

Reply:
> Should this also be done for `block` requests, for example, in `azblob_put_block_request`? Or the metadata will be only supported for `blobs`?

Not sure for now; I will need some time to delve into the documentation.



> I see that in `OSS` the metadata headers is also set for the `append_object_request`.

I remember that only the first append request can include user metadata, but I'm not sure about that. Feel free to do more research on this.